### PR TITLE
Clean out atoms in all cases, also minor fix for render pass tests.

### PR DESCRIPTION
**Pull request by AWoloszyn**

### CODE REVIEW

File: tools/gapit_test_framework.py [hunk 1]
```diff
@@ -362,35 +362,40 @@ def run_test(self, verbose, host, capture_directory):
...
         except GapidUnsupportedException as error:
             print "[ " + "SKIPPED".rjust(10) + " ] " + test_name
             print "     " + error.message
-            return (SKIPPED, error.message)
+            return clear_atoms(SKIPPED, error.message)
         except GapitTestException as error:
             print "[ " + "FAILED".rjust(10) + " ] " + test_name
             print "     " + error.message
-            return (FAILURE, error.message)
+            clear_atoms()
```
Comment (Qining) [inline tools/gapit_test_framework.py]: It seems that this `clear_atoms()` is redundant?

File: tools/gapit_test_framework.py [hunk 2]
```diff
@@ -362,35 +362,40 @@ def run_test(self, verbose, host, capture_directory):
...
+            return clear_atoms(FAILURE, error.message)
         except NamedAttributeError as error:
             exc_type, exc_value, exc_tb = sys.exc_info()
             print "[ " + "FAILED".rjust(10) + " ] " + test_name
             call_site = traceback.format_exception(
                 exc_type, exc_value, exc_tb, limit=2)
             print "    " + error.message + call_site[1]
-            return (FAILURE, error.message)
+            clear_atoms()
```
Comment (Qining) [inline tools/gapit_test_framework.py]: Same as above

File: gapid_tests/command_buffer_tests/BeginAndEndRenderPass_test/BeginAndEndRenderPass_test.py [hunk 3]
```diff
@@ -100,13 +100,6 @@ def expect(self):
         require_equal(0, begin_render_pass_info.clearValueCount)
         require_equal(0, begin_render_pass_info.pClearValues)
 
-        clear_values = VulkanStruct(
-            architecture, CLEAR_COLOR_UNORM_VALUE, get_read_offset_function(
-                begin_render_pass, begin_render_pass_info.pClearValues))
-        # 0.2f == 0x3e4ccccd
-        expected_color_clear_value = [0x3e4ccccd for i in range(4)]
```
Comment (Qining) [inline gapid_tests/command_buffer_tests/BeginAndEndRenderPass_test/BeginAndEndRenderPass_test.py]: Is the hex representation of 0.2f causing problems?

Comment (AWoloszyn) [inline gapid_tests/command_buffer_tests/BeginAndEndRenderPass_test/BeginAndEndRenderPass_test.py]: Nope, pClearValues was changed to be null, so there is no read observation at all.

File: tools/gapit_test_framework.py (hunk 1 above)
Comment (AWoloszyn) [inline tools/gapit_test_framework.py]: oops, that was supposed to go away.

**Pull request merged**